portfolio-project-matching-app/Firebase/clientApp.ts

```diff
@@ -19,5 +19,42 @@ const firebaseApp = initializeApp(firebaseConfig);
 const db = getFirestore();
 // const analytics = getAnalytics(firebaseApp);
 
+
+const getAllDocs = async (collectionName) => {
+  /*
+    DESCRIPTION:  gets all documents from provided collection
+
+    INPUT:        string indicating desired collection name. E.g.,
+                  getAllDocuments('technologies')
+
+    RETURN:       a list of all documents in collection indicated
+  */
+  const coll = collection(db, collectionName);
+  const docSnapshot = await getDocs(coll);
+  const docSnapshotList = docSnapshot.docs.map((doc) => {
+    var arr = doc.data()
+    arr.id = doc.id
+    return arr
+  });
+  return docSnapshotList;
+}
```

Comment:
I was kinda thinking we could create the majority of our queries in `Firebase/clientApp.ts` and import the queries into pages/components as needed. I want to say this is coined the "modular approach" in recent React documentation. So long as we give things self explanatory names, I imagine it would be pretty easy to run with.

Reply:
I like the modular approach!